### DISCUSSION TITLE
Is it possible to trigger onChange outside without using props?

### DISCUSSION THREAD

**Discussion by bsor-dev** [react-hook-form/react-hook-form#9601] [Q&A]
We don't want to use useWatch but instead we want to listen when user pressed the switch button. Also by not using props, is it possible? 

```
<Controller
      control={control}
      name={name}
      render={({field: {value, onChange}}) => {
        return <Switch onChange={value=> onChange(!value)} />;
      }}
    />
```

**Comment by leapful** [comment]
@bsor-dev 

No, it's not possible. 

It would be great if you can explain why not using "onChange" event handler to listen switch button to find another solution for your case.

**Comment by Moshyfawn** [comment]
Well, there's one way to do that: run your `onChange` logic on the form change event:
```jsx
<form onChange={() => /* A FIELD IN THE FORM HAS CHANGED */}></form>
```

But this event is fired when every child input inside the form is changed
